src/FakeItEasy/Expressions/ExpressionArgumentConstraintFactory.cs

```diff
@@ -80,8 +78,20 @@ private static object InvokeExpression(Expression expression)
             return Expression.Lambda(expression).Compile().DynamicInvoke();
         }
 
+        private static void CheckArgumentExpressionIsValid(Expression expression)
+        {
+            var visitor = new ArgumentConstraintExpressionVisitor();
+            visitor.Visit(expression);
+            if (visitor.ConstraintExpressions.Any(e => e != expression))
+            {
+                throw new InvalidOperationException(ExceptionMessages.ArgumentConstraintCannotBeNestedInArgument);

```

Comment:
I'm not sure how easy it is to get a sensible representation of the expression (but I'd think that extracting out the `That`, `_`, or `Ignored` would be sufficient). As for language, if just extracting the member name isn't enough (or isn't feasible), I don't suppose getting the "CLR formatting" is an option with an expression, is it? Like we did in #1047. Perhaps one would have to compile the method to do so?

As a cheap alternative, we could augment the static exception message we have above. Maybe something like 

>An argument constraint, such as That, Ignored, or _, cannot be nested in an argument.

But if we have a `That` inside a `That`, maybe it doesn't help much.